File: src/test/java/com/letgo/book/acceptance/ChangeTitleFeatureTest.java
package com.letgo.book.acceptance;

import org.junit.jupiter.api.Test;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import java.util.UUID;

import static org.junit.jupiter.api.Assertions.*;

final public class ChangeTitleFeatureTest extends TestCase {
    @Test
    public void itShouldChangeABookTitle() {
        UUID id = UUID.randomUUID();
        givenAnExistingBookWith(id.toString(), "OLAKEASE");

        ResponseEntity<String> putResponse = put("/book/" + id + "/title/whatever");
        assertSame(HttpStatus.OK, putResponse.getStatusCode());
        assertFalse(putResponse.hasBody());

        ResponseEntity<String> secondGetResponse = get("/book/" + id);
        assertEquals("whatever", secondGetResponse.getBody());
    }

    @Test
    public void itShouldReturnAnErrorWhenBookIsNotFoundForGivenId() {
        ResponseEntity<String> putResponse = put("/book/" + UUID.randomUUID() + "/title/olakease");
        assertSame(HttpStatus.NOT_FOUND, putResponse.getStatusCode());
        assertFalse(putResponse.hasBody());
    }
}

File: src/test/java/com/letgo/book/acceptance/ChangeTitleFeatureTest.kt
package com.letgo.book.acceptance

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.Test
import org.springframework.http.HttpStatus
import java.util.UUID

private class ChangeTitleFeatureTest : TestCase() {
    @Test
    fun `It should change a book title`() {
        val id = UUID.randomUUID()

        givenAnExistingBookWith(
            id = id.toString(),
            title = "OlaKeAse"
        )

        put(endpoint = "/$id/title/whatever").run {
            assertSame(HttpStatus.ACCEPTED, statusCode)
            assertFalse(hasBody())
        }

        weWaitForMessagesToBeProcessed()

        get(endpoint = "/$id").run {
            assertEquals(
                aJsonBodyWith(
                    """
                        {
                            "id":"$id",
                            "title":"whatever",
                            "edited":true
                        }
                    """
                ),
                body
            )
        }
    }

    @Test
    fun `It should return accepted response even when book is not found for given id`() {
        put(endpoint = "/${UUID.randomUUID()}/title/olakease").run {
            weWaitForMessagesToBeProcessed()
            assertSame(HttpStatus.ACCEPTED, statusCode)
            assertFalse(hasBody())
        }
    }
}
